### DISCUSSION TITLE
Should setting up the hotspot work on Bookworm other than withe NetworkManager?

### DISCUSSION THREAD

**Discussion by stb68** [gitbls/sdm#129] [Q&A]
Hello

Should setting up the hotspot work on Bookworm with hostapd/dnsmasq instead of NetworkManager if you use

   * network
       Args: netman=dhcpcd|noipv6
   * hotspot
       Args: type=routed|wlanip=192.168.254.1|hwmode=gn|channel=10|device=wlan0|country=GB|dhcprange=192.168.254.20,192.168.4.64,255.255.255.0|leasetime=10d|passphrase=1234|ssid=test1|hsmode=dhcpcd|enable=true

I can see that hwmode might need to just be 'g' and maybe the leasetime is invalid (using d for days) but it doesn't seem that hostapd is even installed?

**Comment by gitbls** [comment]
This doesn't work today. The hotspot plugin insists on using network manager on Bookworm. If you look through /etc/sdm/history in the customized IMG you should see:
```
? Plugin hotspot: Only Network Manager supported for hotspot on Bookworm or later
```
Why? Primarily it's to keep the test matrix manageable. Secondarily, network manager is better IMHO than dhcpcd/hostapd in that it's less complex and provides better diagnostic messages in the system journal.

If you want to use dhcpcd/hostapd on Bookworm you can take a private copy of plugins/hotspot and modify it as appropriate to enable dhcpcd/hostapd. Place the modified version (perhaps with a new filename) in /usr/local/sdm/local-plugins and you should be ready to go.

**Aside:** If you're not scanning the logged output in /etc/sdm/history, you may be missing important messages.

**Comment by stb68** [comment]
Thanks for the idea

I did spot that message eventually after posting, but up until then had only noticed messages indicating it had read all the arguments OK and I wasn't 100% sure from the documentation given that you can select dhcpcd for the main networking.

I like your description of the output message prefixes in Issue 125 I found today.  That'll help with the log.

I get your point on NetworkManager, and I'll need to learn it going forward, but I was looking for a quick build at the moment, especially as I already have scripts created to manage the old setup (like changing channels or subnets) and know where to set static leases and whatever else I used before.

My Wireguard VPN stopped working (same Pi as my AP) and none of the various fixes I found online worked, so I was looking for a quick re-build bundled with updating to Bookworm.

I started doing this manually from my notes and found SDM when I was looking for how to remove NetworkManager.  Thought it sounded good, particularly as I need to rebuild my Desktop always on Pi and build a VPN Pi for my Dad as well, and I got side-tracked giving it a try! 

I'll have a look at maybe changing the plugin, test the '.' idea on issue #126 at the same time and have another look at a possible issue with copying to /etc/skel.

Then I'll have to get on with actually building the thing for real!

Thank you for all the efforts.

**Comment by gitbls** [comment]
Sorry I can't help you with WG issues...I'm one step above non-user with WG in that I've installed it and used it a couple of times.

Someday when people run out of sdm issues and requests I'll build a WG plugin because basically all the instructions I see on the net make me want to barf. Edit this file, edit that file, etc etc. Yuck.

> I like your description of the output message prefixes in Issue 125 I found today. That'll help with the log.
I made a note to add this to the documentation somewhere. 

Thanks for your inputs. They help me figure out how I can help you successfully use sdm.

**Reply by stb68** [reply]
Thanks

I used PiVPN (https://pivpn.io/) which does the hard work for you to install and setup either OpenVPN or Wireguard.

I was thinking the other day that it would be great to build on the hotspot plugin idea to add PiVPN and Pi-hole (which I still haven't got around to setting up) plugins.  Of course, you could go on forever and automate everything anybody wants (I need to rebuild my ancient Seafile Pi after all, and then there's my old Pi 2 that has an Access Point and can forward to either wired or wireless main network including a VPN out option, e.g. for sharing a single wifi connection when you're only allowed one client!) and it would become a massive task with any RaspiOS update!

Though WG and Pi-hole would build very nicely on the network and hotspot setup and would probably be fairly widely used.

**Comment by gitbls** [comment]
I agree that they are widely used, and would be helpful to have plugins for them.

The challenge with building plugins is to figure out how the thing that is being automated by the plugin works, what degrees of flexibility would be handy for people to have, and then either wrapping the installer or rewriting it into the plugin.

Both pivpn and pihole have fairly complex installers, and would require a tremendous amount of work to implement and test as a plugin. Not to mention that they are both under active development, so the plugin would need constant care/feeding/testing/etc.

Not to mention...I see many people that have problems with pihole installs. Not sure whether they are caused by pihole or the user, but I sure as heck don't want any of those coming my way!
